augur/curate/apply_geolocation_rules.py

```diff
@@ -111,6 +118,13 @@ def get_annotated_geolocation(geolocation_rules, raw_geolocation, case_sensitive
     # so try to traverse the rules with the next target in raw_geolocation
     if isinstance(current_rules, dict):
         next_traversal_target = raw_geolocation[len(rule_traversal)]
+        if type(next_traversal_target) is not str:
+            # Error out if the next traversal target is not a string
+            raise Exception(
+                f"Expected the next traversal target to be a string, but got {type(next_traversal_target)} instead. " +
+                f"Raw geolocation: {raw_geolocation}, rule traversal: {rule_traversal}. " +
+                f"Current rules: {current_rules}."
```

Comment:
Suggested rewording that points directly to the problem:

```
ERROR: The 'location' field has a null value. Please use an empty string ""
instead of null for missing values.

Input values:
  region: ''
  country: 'Democratic Republic of the Congo'
  division: 'Tshopo,Yahuma'
  location: null
```